Lib/fontTools/misc/configTools.py

```diff
@@ -106,45 +109,71 @@ class Options(Mapping):
     """
 
     __options: Dict[str, Option]
+    __cache: Set[Option]
 
-    def __init__(self) -> None:
+    def __init__(self, other: "Options" = None) -> None:
         self.__options = {}
+        self.__cache = set()
+        if other is not None:
+            for option in other.values():
+                self.register_option(option)
 
     def register(
         self,
         name: str,
         help: str,
         default: Any,
         parse: Callable[[str], Any],
-        validate: Callable[[Any], bool],
+        validate: Optional[Callable[[Any], bool]] = None,
     ) -> Option:
-        """Register a new option."""
-        return self.register_option(name, Option(help, default, parse, validate))
+        """Create and register a new option."""
+        return self.register_option(Option(name, help, default, parse, validate))
 
-    def register_option(self, name: str, option: Option) -> Option:
+    def register_option(self, option: Option) -> Option:
         """Register a new option."""
+        name = option.name
         if name in self.__options:
             raise ConfigAlreadyRegisteredError(name)
+        # sanity check option values are unique
+        assert option not in self.__cache
         self.__options[name] = option
+        self.__cache.add(option)
         return option
 
-    def __getitem__(self, __k: str) -> Option:
-        return self.__options.__getitem__(__k)
+    def is_registered(self, option: Option) -> bool:
+        """Return True if the option object is already registered."""
+        return option in self.__cache
```

Comment:
you could look for the name in the __options

Reply:
same name is no guarantee that two option objects are the same thing. One could make a new Option object with the same name as one already registered (but e.g. different default) and it would pass as if it were registered if we only checked its name.

Reply:
That feels like adding weird semantics to what the "identity" of an option is. I would rather implement it as just an extra check like this:

```python
class Config:
    __options = {}

    def check_option_is_same(option: Option):
         assert self.__options[option.name] is option

     # and then call the check above anytime you're given an Option instance
     # the check is O(1)
     # no need for an extra set
```